sites/all/modules/reol_app_feeds/src/Helper/NodeHelper.php

```diff
@@ -0,0 +1,313 @@
+
+  /**
+   * Get value of a field.
+   *
+   * If the requested field does not exist, NULL will be returned.
+   *
+   * Note: This function is very similar to what the `entity_metadata_wrapper`
+   * provides, but this function handles non-existing fields gracefully.
+   *
+   * @param object $entity
+   *   The entity (a node or a \ParagraphsItemEntity).
+   * @param string $field_name
+   *   The field names.
+   * @param string|null $sub_field_name
+   *   The optional sub-field names.
+   * @param bool $multiple
+   *   If set, multiple field values will be returned as an array. Otherwise,
+   *   only the first value will be returned.
+   *
+   * @return array|mixed|null
+   *   The field value.
+   */
+  public function getFieldValue($entity, $field_name, $sub_field_name = NULL, $multiple = FALSE) {
+    if (!isset($entity->{$field_name}[LANGUAGE_NONE])) {
+      return NULL;
+    }
+
+    $values = $entity->{$field_name}[LANGUAGE_NONE];
+
+    if (NULL !== $sub_field_name) {
+      $values = array_column($values, $sub_field_name);
+    }
+
+    return $multiple ? $values : reset($values);
+  }
+
+  /**
+   * Get text value of a field.
+   *
+   * @param object $entity
+   *   The entity.
+   * @param string $field_name
+   *   The field names.
+   * @param string|null $sub_field_name
+   *   The optional sub-field names.
+   * @param bool $multiple
+   *   If set, multiple field values will be returned as an array. Otherwise,
+   *   only the first value will be returned.
+   *
+   * @return array|mixed|null
+   *   The text field value.
+   */
+  public function getTextFieldValue($entity, $field_name, $sub_field_name = NULL, $multiple = FALSE) {
+    $values = $this->getFieldValue($entity, $field_name, $sub_field_name, TRUE);
+    $values = array_map([$this, 'getTextValue'], $values);
+
+    return $multiple ? $values : reset($values);
+  }
+
+  /**
+   * Get list of paragraphs on a node.
+   *
+   * @param object $entity
+   *   The entity.
+   *
+   * @return array|\ParagraphsItemEntity[]
+   *   The paragraphs.
+   */
+  public function getParagraphs($entity) {
+    $paragraphs = [];
+
+    $helper = new ParagraphHelper();
+
+    $fields = $helper->getParagraphFields($entity);
+    foreach ($fields as $field_name => $field) {
+      $paragraphs += $helper->loadParagraphs($entity, $field_name);
+    }
+
+    return $paragraphs;
+  }
+
+  /**
+   * Get text value.
+   *
+   * @param mixed $value
+   *   The value.
+   *
+   * @return string|null
+   *   The text value if any.
+   */
+  private function getTextValue($value) {
+    return isset($value['safe_value']) ? $value['safe_value'] : NULL;
+  }
+
+  /**
+   * Get body text value from a node.
+   *
+   * @param object $node
+   *   The node.
+   *
+   * @return string|null
+   *   The body text value if any.
+   */
+  public function getBody($node) {
+    return $this->getTextFieldValue($node, 'body', NULL, FALSE);
+  }
+
+  /**
+   * Get image url.
+   *
+   * @param mixed $value
```

Comment:
Should the type have been changed to `array|null` ?

Reply:
Yes.